src/elements/Asset.php

```diff
@@ -1493,33 +1494,54 @@ public function getAdditionalButtons(): string
+                    .attr('title', result.formattedSizeInBytes);
+                let \$dimensionsVal = $('#dimensions-value');
+                if (result.dimensions) {
+                    if (!\$dimensionsVal.length) {
+                        $(
+                            '<div class="data">' +
+                            '<dt class="heading">$dimensionsLabel</div>' +
+                            '<dd id="dimensions-value" class="value"></div>' +
+                            '</div>'
+                        ).appendTo($('#details > .meta.read-only'));
+                        \$dimensionsVal = $('#dimensions-value');
+                    }
+                    \$dimensionsVal.text(result.dimensions);
+                } else if (\$dimensionsVal.length) {
+                    \$dimensionsVal.parent().remove();
+                }
+                $updatePreviewThumbJs
+                Craft.cp.runQueue();
+                if (result.error) {
                     $('#thumb-container').removeClass('loading');
-                    alert(data.result.error);
+                    alert(result.error);
                 } else {
-                    $('#new-filename').val(data.result.filename);
-                    $('#file-size-value')
-                        .text(data.result.formattedSize)
-                        .attr('title', data.result.formattedSizeInBytes);
-                    let \$dimensionsVal = $('#dimensions-value');
-                    if (data.result.dimensions) {
-                        if (!\$dimensionsVal.length) {
-                            $(
-                                '<div class="data">' +
-                                '<dt class="heading">$dimensionsLabel</div>' +
-                                '<dd id="dimensions-value" class="value"></div>' +
-                                '</div>'
-                            ).appendTo($('#details > .meta.read-only'));
-                            \$dimensionsVal = $('#dimensions-value');
-                        }
-                        \$dimensionsVal.text(data.result.dimensions);
-                    } else if (\$dimensionsVal.length) {
-                        \$dimensionsVal.parent().remove();
-                    }
-                    $updatePreviewThumbJs
-                    Craft.cp.runQueue();
+
                 }
-            }
+            },
+            fileuploadfail: (event, data) => {
+                const response = event instanceof Event
+                    ? event.detail
+                    : data?.jqXHR?.responseJSON;
+                
+                let {message, filename} = response || {};
+                
+                if (!message) {
+                    message = filename
+                        ? Craft.t('app', 'Replace file failed for “{filename}”.', {filename})
+                        : Craft.t('app', 'Replace file failed.');
+                }
+                
+              Craft.cp.displayError(message);
+            },
```

Comment:
Replaced the vanilla js `alert` with `Craft.cp.displayError` while I was in there